src/backend/base/langflow/components/tools/file_manipulation.py

```
@@ -0,0 +1,1301 @@
+            A tuple containing the modified content and a list of (start, end, line number) for each replacement made.
+        
+        Raises:
+            ValueError: If an invalid regular expression is provided.
+        """
+        lines = content.split('\n')
+        occurrences = []
+        
+        if use_regex:
+            # Regex-based search
+            try:
+                regex = re.compile(search, re.MULTILINE | re.DOTALL)
```

Ruff (0.11.9)

Local variable `re` referenced before assignment

(F823)